src/preamble/01_what_do_we_mean_when_we_say_cabal.md

```diff
@@ -1 +1,73 @@
 # What do we mean when we say cabal?
+
+Cabal is an umbrella term that can refer to either cabal-the-spec (.cabal
+files), cabal-the-library (code that understands .cabal files), or
+cabal-the-tool (the cabal-install package which provides the cabal executable).
+
+## tl;dr
+
+1. [cabal-install](#cabalInstall) is the command line utility to help configure,
+   compile and install Haskell libraries and programs.
+
+2. [The Cabal build system](#cabailBuildSystem), a specification for defining
```

Comment:
I'll change Cabal build to CABAL build. I can mention "Cabal can take a haskell dependency graph (of external and internal modules) and use GHC to build it." in the details section?

Reply:
@JonathanLorimer  what would be better? adding it here in tldr section or in the more details section?

Reply:
@srirajshukla I am okay with you mentioning the dependency graph bit in the details section. But I still think you need to be careful about using the word `specification`. It could be confused with the `.cabal` file being a specification for your haskell package. Also I don't think `Cabal` is really a specification, it is really just a library and a build system.